tests/test_browser.py

```diff
@@ -94,15 +94,18 @@ def test_prepare_request_file():
     form.find("input", {"name": "pic"})["value"] = pic_path
 
     browser = mechanicalsoup.Browser()
-    request = browser._build_request(form)
-
-    assert request.files
-    assert request.files["pic"]
-    assert request.files["pic"].read() == ":-)".encode("utf8")
-    assert "pic" not in request.data
-
-    request = browser._prepare_request(form)
-    assert "multipart/form-data" in request.headers["Content-Type"]
+    response = browser._request(form)
+
+    # Check the response only "files" includes a "pic" keyword
+    data = response.json()
+    for key in data.keys():
+        if key == "files":
+            assert data[key]["pic"] == ":-)"
+            continue
+        with pytest.raises(Exception):
+            data[key]["pic"]
```

Comment:
The 2 lines above sound like a complex way to write

    assert "pic" not in data[key]

Reply:
I wish it was, but because `data[key]` can be both a dict and `None`, you get both `IndexError` and `TypeError` when you try to use an iterator (e.g. `in`) with it.

Reply:
OK, but then I'd rather make this explicit in the code with stg like `assert data[key] is None or "pic" not in data[key]`. That would make it clear to the reader what the possible values are, and would avoid this `with pytest.raises(Exception):` : not really dangerous here, but it's a construct in which any buggy code that raise an exception makes the test pass...

Reply:
Good point! How does it look now?

Reply:
Good.